### ISSUE TITLE
Expose `architecture` from `Abi` object

### ISSUE THREAD

**Issue by miquelbeltran** [dart-lang/sdk#56521]
Hello Dart team!

Currently, the easiest way to obtain the compiled architecture of an app is to call to `Abi.current().toString()` and then parse the `String`, e.g.

```
final abi = Abi.current().toString();
final architecture = abi.split('_').last;
```

Could it be possible to expose the `Architecture` property directly?

Related ticket: https://github.com/dart-lang/sdk/issues/40231

**Comment by lrhn** [comment]
As Josh Bloch said in Effective Java, don't provide information in `toString` that you don't provide otherwise. People *will* parse `toString`, and then its format becomes part of your API.

**Comment by miquelbeltran** [comment]
Even the Dart team does it ;)

https://github.com/dart-lang/native/blob/4a18bf2964b69075d9ec3ec1295d4985494741bb/pkgs/native_assets_cli/test/api/target_test.dart#L18

**Comment by lrhn** [comment]
There's probably a reason he said that - everybody does it, until they learn why not to, often the hard way.

**Comment by dcharkes** [comment]
Hi @miquelbeltran!

I guess the main question is whether such functionality should live in the Dart SDK, or in a package.
And if it lives in a package, whether it should be a different package than `native_assets_cli`: https://pub.dev/documentation/native_assets_cli/latest/native_assets_cli/Architecture-class.html

Also, if we do `Architecture`, we should also talk about `OS`.
We also don't have an `OS` in `dart:`. And the `Platform` can only talk about the current OS.
`package:platform` also doesn't have an `OS`, but it can use `FakePlatform` to talk about a different OS.
Because `package:platform` is already the way to deal with OS (not the cleanest, but okay), I don't think we'll ever add `OS` to `dart:` (I believe I opened an issue about this years back, but i can't find it anymore.)

Pros for package:

* Smaller SDK
* Easier to iterate
* OS will also not be in `dart:`

Pros for `dart:`

* No rolling when `Abi`s are added

I think I'm leaning towards a package instead of `dart:`.

Even though `package:native_assets_cli` is meant for writing build hooks, nothing prevents you from using that package for the `Architecture` class in other contexts.

**Comment by miquelbeltran** [comment]
Hi Daco!

My pros for having that in the SDK itself is discoverability and one less 3rd party dependency in code.

`native_assets_cli` is currently listed as experimental, so I was a bit wary about adding it to my project.

Maybe the team would be open to extract the `Architecture` and `OS` parts from it and put it in a separate package under `dart-lang/native/pkgs`?

**Comment by dcharkes** [comment]
> `native_assets_cli` is currently listed as experimental, so I was a bit wary about adding it to my project.

I don't believe we'd change the API of that class any time. (We are however going to change the package into `package:hook`.)

> Maybe the team would be open to extract the `Architecture` and `OS` parts from it and put it in a separate package under `dart-lang/native/pkgs`?

That's an option. Something like `package:native_core`, or even `package:native`. Where we move the most central concepts. But I don't want to do that right now while many things are in flux. While these packages are under development every release cycle requires publishing them in sequence, and it would add yet another step.
